src/badges.js

```diff
@@ -707,6 +732,71 @@ function findPNASDOIs () {
   return els
 }
 
+/**
+ * findConnectedPapersDOIs looks in reference tags that link to doi.org.
+ * @returns {Array<{ citeEl: Element, doi: string}>} - Return
+ */
+function findConnectedPapersDOIs () {
+  const els = []
+  const cites = document.body.querySelectorAll('#open-in-container')
+  for (const cite of cites) {
+    const anchors = cite.querySelectorAll('a')
+    for (const anchor of anchors) {
+      if (anchor?.href?.match(/doi\.org\/(.+)/)) {
+        els.push({
+          citeEl: cite,
+          doi: anchor.href.match(/10\..*/)[0]
+        })
+      }
+    }
+  }
+  return els
+}
+
+/**
+ * findPeerJDOIs looks in reference tags that link to doi.org.
+ * @returns {Array<{ citeEl: Element, doi: string}>} - Return
+ */
+function findPeerJDOIs () {
+  const els = []
+  const cites = document.body.querySelectorAll('.citation')
+  for (const cite of cites) {
+    const anchors = cite.querySelectorAll('a')
+    for (const anchor of anchors) {
+      const doi = anchor?.href?.match(/10\.(.+)/)
+      if (doi) {
+        els.push({
+          citeEl: cite,
+          doi: decodeURIComponent(doi[0])
+        })
+      }
+    }
+  }
+  return els
+}
+
+/**
+ * findClinicalTrialsDOIs looks in reference links for doi.org.
+ * @returns {Array<{ citeEl: Element, doi: string}>} - Return
+ */
+function findClinicalTrialsDOIs () {
+  const els = []
+  const cites = document.body.querySelectorAll('a')
+  for (const cite of cites) {
+    const re = /(10.\d{4,9}\/[-._;()/:A-Z0-9]+)/ig
```

Comment:
Should this be earlier (top level) to prevent it from re-compiling each time or is it smart enough to only do it once?

Reply:
ya thats a good idea